x/evm/keeper/grpc_query.go

```diff
@@ -217,6 +217,10 @@ func (k Keeper) EthCall(c context.Context, req *types.EthCallRequest) (*types.Ms
 	}
 
 	ctx := sdk.UnwrapSDKContext(c)
+	if len(req.ChainId) > 0 {
```

Comment:
We should check that the chain ID is empty in the context too

Reply:
Do you mean if the chain id is empty in the context, we should ignore the request?